libraries/AP_Beacon/AP_Beacon.cpp

```diff
@@ -225,6 +225,73 @@ uint32_t AP_Beacon::beacon_last_update_ms(uint8_t beacon_instance) const
     return beacon_state[beacon_instance].distance_update_ms;
 }
 
+// create fence boundary points
+bool AP_Beacon::update_boundary_points()
+{
+    // we need three points at least to create fence
+    // if already tried to create boundary, return false
+    if (!device_ready() || num_beacons < AP_BEACON_MINIMUM_FENCE_BEACONS || boundary_num_beacons == num_beacons) {
+        return false;
+    }
+
+    // delete to recreate boundary
+    if (_boundary != nullptr) {
+        delete _boundary;
+        _boundary = nullptr;
+    }
+
+    // check if we need to create array
+    _boundary = (Vector2f *)calloc(1, (num_beacons + 1) * sizeof(Vector2f));
+
+    if (_boundary == nullptr) {
+        return false;
+    } else {
+        boundary_num_beacons = num_beacons;
+    }
+
+    // accumulate beacon points
+    for (uint8_t index = 0; index < num_beacons; index++) {
+        // if a beacon is not healthy, delete _boundary to renew array
+#if CONFIG_HAL_BOARD != HAL_BOARD_SITL
```

Comment:
@OXINARF, Thank you for reviewing. When I tesed in SITL, beacon_healthy(index) was false. I thouhgt it depends on my PC spec. I will check it again.